### PR TITLE
Handle top level options set in subprojects. Closes #13847.

### LINKED ISSUE #13847: pkgconfig.relocatable in subproject default_options fails in parent project setup
**Describe the bug**
When setting up a parent project that includes a subproject with `pkgconfig.relocatable=true` in its `default_options`, meson setup fails in the parent project with the error:
```
subprojects/mysubprj/meson.build:1:0: ERROR: In subproject mysubprj: Unknown options: "mysubprj:pkgconfig.relocatable"
```

However, running meson setup directly within the subproject works without issue.
This change in behavior started with the following commit:
```
commit 6e200222957063819a00e3bf767ce28b7489c31f
Author: Jussi Pakkanen <jpakkane@gmail.com>
Date:   Sun Jul 14 19:33:41 2024 +0300

    Remove module type from OptionKey.
```

**To Reproduce**
In the parent project `meson.build`:
```
project('test', version : '1.0')
subproject('mysubprj')
```

In the subproject `meson.build`:
```
project('mysubprj', version : '1.0', default_options: ['pkgconfig.relocatable=true'])
```

Command-line steps:
```sh
cd test
meson setup build
# Error as shown above
cd test/subprojects/mysubprj
meson setup build
# Success
```

**Expected behavior**
The meson setup command in the parent project should successfully handle `pkgconfig.relocatable=true` in subproject `default_options` as it did in previous versions.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
native
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Ubuntu 24.04
* what Python version are you using e.g. 3.8.0
3.12.3
* what `meson --version`
1.6.0

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: test cases/unit/123 pkgsubproj/meson.build [hunk 1]
```diff
@@ -0,0 +1,3 @@
+project('pkg_opt_test', 'c')
```
Comment (eli-schwartz) [inline test cases/unit/123 pkgsubproj/meson.build:1]: We can omit the language here and avoid performing compiler initialization for this unittest.

File: unittests/linuxliketests.py [hunk 2]
```diff
@@ -1863,3 +1863,8 @@ def test_complex_link_cases(self):
         self.assertIn('build t9-e1: c_LINKER t9-e1.p/main.c.o | libt9-s1.a libt9-s2.a libt9-s3.a\n', content)
         self.assertIn('build t12-e1: c_LINKER t12-e1.p/main.c.o | libt12-s1.a libt12-s2.a libt12-s3.a\n', content)
         self.assertIn('build t13-e1: c_LINKER t13-e1.p/main.c.o | libt12-s1.a libt13-s3.a\n', content)
+
+    def test_sp_options(self):
```
Comment (eli-schwartz) [inline unittests/linuxliketests.py:1867]: I wonder if this makes more sense in unittests/platformagnostictests.py :)

File: mesonbuild/coredata.py [hunk 3]
```diff
@@ -658,9 +658,20 @@ def set_options(self, opts_to_set: T.Dict[OptionKey, T.Any], subproject: str = '
             elif k.machine != MachineChoice.BUILD and not self.optstore.is_compiler_option(k):
                 unknown_options.append(k)
         if unknown_options:
-            unknown_options_str = ', '.join(sorted(str(s) for s in unknown_options))
-            sub = f'In subproject {subproject}: ' if subproject else ''
-            raise MesonException(f'{sub}Unknown options: "{unknown_options_str}"')
+            if subproject:
+                # The subproject may have top-level options that should be used
+                # when it not a subproject. Ignore those for now. With option
```
Comment (eli-schwartz) [inline mesonbuild/coredata.py:663]:
```suggestion
                # when it is not a subproject. Ignore those for now. With option
```

File: test cases/unit/123 pkgsubproj/meson.build (hunk 1 above)
Comment (eli-schwartz) [reply test cases/unit/123 pkgsubproj/meson.build:1]: (and in the subproject)

File: unittests/linuxliketests.py (hunk 2 above)
Comment (jpakkane) [reply unittests/linuxliketests.py:1867]: This only makes sense when pkg-config is installed.  Pure Visual Studio and pure Xcode do not have it.

Comment (eli-schwartz) [reply unittests/linuxliketests.py:1867]: We are testing top-level options, not pkgconfig per se. Also, the pkgconfig *module* doesn't care whether pkg-config programs are installed or not :) it is purely about writing out the text file format.

Comment (jpakkane) [reply unittests/linuxliketests.py]: Renamed. I'm not going to spend time moving this around. It is sufficient that it is run in _some_ environment. Let's not bikeshed this to death but instead get the release out.

**Pull request merged**